core/src/test/java/org/apache/brooklyn/core/typereg/RegisteredTypesTest.java

```diff
@@ -0,0 +1,123 @@
+package org.apache.brooklyn.core.typereg;
+
+import org.apache.brooklyn.api.typereg.BrooklynTypeRegistry;
+import org.apache.brooklyn.api.typereg.RegisteredType;
+import org.testng.annotations.Test;
+
+import javax.annotation.Nullable;
+
+import static org.testng.Assert.*;
+
+public class RegisteredTypesTest {
+
+    public static final String DRBD_YAML = "brooklyn.catalog:\n" +
```

Comment:
It's hard to tell when reading the code what the significant differences are between these various plans - think it would be worth cutting them down to something smaller where you could read the code and see more easily what comparison you can make.

Reply:
also they would be easier to read if you used `Strings.lines()` to join them

Reply:
happy to merge as is but i like @geomacy 's suggestion readability could be improved.  maybe an  inner static class `EquivalentPlanGenerator` that has fields `usesQuotes1` and `includeComments` so code is like:

    plan +=
      "    drbdIconUrl: "+quoteIfTrue(usesQuotes1, "https://s3.eu-central-1.amazonaws.com/misc-csft/drbd.jpg");

then each test might be rewritten

    assertTrue(RegisteredTypes.arePlansEquivalent(
        new EquivalentPlanGenerator().usesQuotes1(true).build(),
        new EquivalentPlanGenerator().includesComments1(true).build() ));

(but as i said not at all necessary -- nice job adding the tests!)